vm-migration/src/lib.rs

```diff
@@ -106,23 +88,10 @@ impl SnapshotData {
     where
         T: Serialize,
     {
-        let data = serde_json::to_vec(state)
+        let state = serde_json::to_string(state)
             .map_err(|e| MigratableError::Snapshot(anyhow!("Error serialising: {}", e)))?;
 
-        Ok(SnapshotData(data))
-    }
-
-    /// Create from versioned state
-    pub fn new_from_versioned_state<T>(state: &T) -> Result<Self, MigratableError>
-    where
-        T: Versionize + VersionMapped,
-    {
-        let mut data = Vec::new();
-        state
-            .serialize(&mut data, &T::version_map(), VMM_VERSION)
-            .map_err(|e| MigratableError::Snapshot(anyhow!("Error serialising: {}", e)))?;
-
-        Ok(SnapshotData(data))
+        Ok(SnapshotData { version: 1, state })
```

Comment:
> Nit: I think it is better to define a constant for the `snapshot_version`.
> 
> Also, I am thinking we may want to provide both `snapshot_major_version` and `snapshot_minor_version`. With this, we can make it clear that a major version bump indicates breaking live upgrade, and we can put checks in place to explicitly abort live upgrades from incompatible snapshots. Meanwhile, minor version bumps can be used for potential manual fix on live upgrades during a transitioning period of time.

I wanted to make the version semantic and separate from the the CH version - since it's more about the the contents changing. Adaption to the version would need to be local e.g. in the place that uses the snapshot data - it therefore would be best to know about the version.

Also I didn't want to use the CH version as that is something we would have to update on release - ideally we would never use this version field as we would agree to not break the updates any further.

I only chose `1` vs `0` as the value so that it would not be the default value for u64.

Reply:
Maybe I was not clear. I totally agree the idea of not using CH versions as the snapshot/snapshotdata versions. My comment was about defining a constant for it for better readability and using both major and minor versions for the case I mentioned above (mostly for explicitly aborting live-upgrades across breaking versions).